pkg/controller/job/job_controller.go

```diff
@@ -843,6 +843,7 @@ func (jm *Controller) manageJob(job *batch.Job, activePods []*v1.Pod, succeeded
 					if completionIndex != unknownCompletionIndex {
 						template = podTemplate.DeepCopy()
 						addCompletionIndexAnnotation(template, completionIndex)
+						template.Spec.Hostname = fmt.Sprintf("%s-%d", job.Name, completionIndex)
```

Comment:
is this valid for all combinations of valid job names and valid completionIndex?

Reply:
We don't currently validate that the job name is a valid dns name, we should do that for jobs with `CompletionMode= IndexedCompletion`. @alculquicondor is there a maximum index? we should subtract the length from the max allowed DNs subdomain length like we do for cron job: https://github.com/kubernetes/kubernetes/blob/aa450787bff54c96b4588e7ddbfdf9bda89a882a/pkg/apis/batch/validation/validation.go#L208.

Reply:
The completions field is an `int32`, so 10 would be the max number of digits.

@liggitt can we tighten the validation for an alpha feature?

Reply:
10 + 1 for the dash, right?

you can do the validation when creating an indexed job, and when updating a non-indexed job to an indexed job

Reply:
completionMode is immutable :)

Reply:
even better! validating on creation sgtm.

Reply:
validation added in https://github.com/kubernetes/kubernetes/pull/101292